lambdas/check-existing-identity/src/main/java/uk/gov/di/ipv/core/checkexistingidentity/CheckExistingIdentityHandler.java

```diff
@@ -449,19 +458,28 @@ private JourneyResponse buildReuseResponse(
                     : JOURNEY_OPERATIONAL_PROFILE_REUSE;
         }
 
+        var credentials = VcHelper.filterVCBasedOnProfileType(vcs, attainedVot.getProfileType());
         sessionCredentialsService.persistCredentials(
-                VcHelper.filterVCBasedOnProfileType(vcs, attainedVot.getProfileType()),
-                auditEventUser.getSessionId(),
-                false);
+                credentials, auditEventUser.getSessionId(), false);
+        migrateCredentialsToEVCS(auditEventUser.getUserId(), credentials);
 
         return JOURNEY_REUSE;
     }
 
+    private void migrateCredentialsToEVCS(String userId, List<VerifiableCredential> credentials)
+            throws VerifiableCredentialException {
+        if (configService.enabled(EVCS_WRITE_ENABLED)) {
+            evcsMigrationService.migrateExistingIdentity(userId, credentials);
+            credentials.forEach(credential -> credential.setMigrated(Instant.now()));
+            verifiableCredentialService.updateIdentity(credentials);
```

Comment:
I wonder if this logic should also live inside `EvcsMigrationService`, so you can't migrate without doing it? Not super important